wetmap/src/components/googleMap/view.tsx

```diff
@@ -0,0 +1,359 @@
+import React from 'react';
+import {
+  GoogleMap,
+  Marker,
+  HeatmapLayer,
+} from '@react-google-maps/api';
+import style from './style.module.scss';
+import { Cluster, TempMarker } from './types';
+import useSupercluster from 'use-supercluster';
+import anchorClusterIcon from '../../images/mapIcons/AnchorCluster.png';
+import anchorIconGold from '../../images/mapIcons/AnchorGold.png';
+import mantaIcon from '../../images/Manta32.png';
+import {
+  useMemo,
+  useState,
+  useContext,
+  useEffect,
+  useLayoutEffect,
+} from 'react';
+import { CoordsContext } from '../contexts/mapCoordsContext';
+import { ZoomContext } from '../contexts/mapZoomContext';
+import { DiveSitesContext } from '../contexts/diveSitesContext';
+import { AnimalContext } from '../contexts/animalContext';
+import { MasterContext } from '../contexts/masterContext';
+import { MinorContext } from '../contexts/minorContext';
+import { PinSpotContext } from '../contexts/pinSpotContext';
+import { SelectedDiveSiteContext } from '../contexts/selectedDiveSiteContext';
+import { SelectedShopContext } from '../contexts/selectedShopContext';
+import { HeatPointsContext } from '../contexts/heatPointsContext';
+import { MapBoundsContext } from '../contexts/mapBoundariesContext';
+import { DiveSpotContext } from '../contexts/diveSpotContext';
+import { ShopModalContext } from '../contexts/shopModalContext';
+import { SitesArrayContext } from '../contexts/sitesArrayContext';
+import { ZoomHelperContext } from '../contexts/zoomHelperContext';
+import { CarrouselTilesContext } from '../contexts/carrouselTilesContext';
```

Comment:
Do you think it'd be possible to access context in index.tsx and pass required data as props to here?
I understand that googleMap is not an ordinary component, and I'm not sure if that approach going to work here but still it would be nice to have independent(dumb) view, the ability to test it in storybook...

Reply:
Its a big job but I have started it< however i will purposely leave some in view for now, the are the ones with the "master" and "minor" context references, these will be deprecated once the trip creator is built in. I have a much better way to handle the map flip scenarios in the mobile version that I will being over when the time comes so I want the affected items isolated